File: Dante1/Module1.07/Zadanie7.10/main.c
#include <stdio.h>

float quadratic_equation(float a, float b, float c, float x)
{
    return a * x * x + b * x + c;
}

int main()
{
    float a, b, c, x;

    printf("Enter: ");

    int check1 = scanf("%fx^2%fx%f", &a, &b, &c);
    if (check1 != 3)
    {
        printf("\nIncorrect input\n");
        return 1;
    }

    printf("\nEnter x: ");

    int check2 = scanf("%f", &x);
    if (check2 != 1)
    {
        printf("\nIncorrect input\n");
        return 1;
    }

    float result = quadratic_equation(a, b, c, x);
    printf("\nResult: %f\n", result);

    return 0;
}

File: Dante1/Module1.07/Zadanie7.10/main.rs
fn quadratic_equation(a: f32, b: f32, c: f32, x: f32) -> f32 {
    a * x * x + b * x + c
}

fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let mut numbers = input.split_whitespace();

    let num1: f32 = numbers.next().expect("Нет первого числа").parse().expect("Ошибка парсинга");
    let num2: f32 = numbers.next().expect("Нет второго числа").parse().expect("Ошибка парсинга");
    let num3: f32 = numbers.next().expect("Нет третьего числа").parse().expect("Ошибка парсинга");

    input.clear();

    std::io::stdin().read_line(&mut input).expect("err");
    let value: f32 = input.trim().parse().expect("err");

    println!("Result is: {}", quadratic_equation(num1, num2, num3, value));
}
